### PR TITLE
Hcap 699 post hire withdrawal display

### PR DESCRIPTION
**Pull request by Zenen-freshworks**
The base functionality for 699 was already approved; this PR is being reopened to resolve bugs that came up when it was merged.

### CODE REVIEW

File: client/Dockerfile.dev [hunk 1]
```diff
@@ -1,10 +1,10 @@
-FROM node:14-alpine
+FROM public.ecr.aws/bitnami/node:14.17.3
 
 # Allow node modules to be called directly
 ENV PATH $PATH:/client/node_modules/.bin
 
 # Build client
-RUN apk add --no-cache git python g++ make
+# RUN apk add --no-cache git python g++ make
```
Comment (fwpushan) [inline client/Dockerfile.dev:7]: Remove this line, this is pure overhead.

File: client/src/pages/private/SiteParticipantsTable.js [hunk 2]
```diff
@@ -150,8 +161,8 @@ export default ({ id, siteId, onArchiveParticipantAction }) => {
             participantName: `${row.participantJoin.body.firstName} ${row.participantJoin.body.lastName}`,
             hiredDate: row.data.hiredDate,
             startDate: row.data.startDate,
-            withdrawnDate: row.created_at,
-            reason: row.data.final_status,
+            withdrawnDate: row.data.endDate,
```
Comment (fwpushan) [inline client/src/pages/private/SiteParticipantsTable.js:164]: possibly data is integral part of row object, but for safety we can use optional chaining.
like ``
```suggestion
            withdrawnDate: row.data?.endDate,
```

File: Dockerfile [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
+FROM public.ecr.aws/bitnami/node:14.17.3 AS server
 
 # Static env vars
 ARG VERSION
 ENV VERSION $VERSION
 ENV NODE_ENV production
 
 # Configure server
-RUN apk add --no-cache git
+# RUN apk add --no-cache git
```
Comment (fwpushan) [inline Dockerfile:21]: Remove this line

File: server/services/participants.js [hunk 4]
```diff
@@ -121,25 +121,26 @@ const getWithdrawnParticipantsBySite = async (siteID) => {
...
+  // a given site, we need to match the statuses created by a specific
+  // employer, as the withdrawl action creates two current statuses
+  // simultaneously. One holds the siteID, the other holds the end date and
+  // reason for withdrawal. We match them based on employerID.
+  const employerIDs = participants
+    .filter((ppt) => ppt.data?.previousData?.site === parseInt(siteID, 10))
+    .map((ppt) => ppt.employer_id);
+
+  return participants.filter(
+    (ppt) =>
```
Comment (fwpushan) [inline server/services/participants.js:139]: ppt is not very appropriate naming here, `participant` would be ideal.

**Review by fwpushan — COMMENTED** [review]
Overall LGTM, only few suggestions.

**Review by dbayly-freshworks — APPROVED** [review]
Looks ok to me

**Review by fwpushan — APPROVED** [review]
LGTM

**Review by dbayly-freshworks — APPROVED** [review]

**Review by fwpushan — APPROVED** [review]

**Pull request merged**